packages/lib-runtime/src/utils/fetch.ts

```diff
@@ -0,0 +1,26 @@
+import { CustomError } from './errors';
+
+/**
+ * Fetch a resource via HTTP protocol and return the response body text.
+ */
+export type FetchResource = (url: string, method: string) => Promise<string>;
+
+class HTTPError extends CustomError {
+  constructor(message: string, readonly status: number, readonly response: Response) {
+    super(message);
+  }
+}
+
+/**
+ * Basic implementation of `FetchResource` that uses the `fetch` API.
+ */
+export const basicFetch: FetchResource = async (url, method) => {
+  const response = await fetch(url, { method });
+
+  if (response.ok) {
+    const responseText = await response.text();
```

Comment:
Intent is to have a JSON Fetch? I'd expect very rarely are we wanting raw text.

Reply:
@andrewballantyne The intent here was that plugin SDK (its "runtime" library specifically) would work with a generic `FetchResource` compatible function, allowing to fetch arbitrary resources over HTTP and return their contents as plain text.

```ts
export type FetchResource = (url: string, method: string) => Promise<string>;
```

The `basicFetch` implementation is used as a default fallback. Specific host apps, like Console, will probably want to provide a custom implementation, like Console's `coFetch` which adds some HTTP headers and has retry & timeout logic built into it.

For now, `FetchResource` is used only in `PluginLoader.getPluginManifest`.

Is there a specific reason for doing `response.json()` instead of calling `JSON.parse` on `response.text()`? (The latter allows us to have a single `FetchResource` interface for fetching & processing all kinds of resources, not only JSON.)

Reply:
No this is fair... this was before your rundown of the code.